torchtune/modules/loss/loss_types.py

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from abc import ABC, abstractmethod
+
+import torch
+from torch import nn
+
+
+class SFTLoss(ABC):
```

Comment:
What's our plan for non-SFT losses? E.g. for forward KL loss as used in KD -- do we need to define another abstract class for those too? (Also nit but why not protocol?)

Reply:
Yes, we have to define one per recipe type at this point. I'd be open to finding a different access to define these on but this is what there was and it seems reasonable.

re net: protocols don't work well for interfaces. They're very inefficient for runtime checks and not as clear a definition for users to extend (ABCs can be traced directly by the user: see an example class -> go to abc to see how to write your own). At this point I'm in favor of phasing protocols out of torchtune.